### PR TITLE
Refs [TASK][36326] : Create Favorite Screen

### PR DESCRIPTION
**Pull request by hideoncalm**
## Related Tickets
- [#Task 36326: Create Favorite Screen](https://edu-redmine.sun-asterisk.vn/issues/36326)
## WHAT
- 
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/coinqapp/ui/coindetail/CoinDetailFragment.kt [hunk 1]
```diff
@@ -68,6 +97,30 @@ class CoinDetailFragment : BaseFragment<FragmentCoinDetailBinding>() {
...
     }
 
+    override fun showCoinFavorite(isFavorite: Int) {
+        val check = (isFavorite >= 1)
+        binding.buttonFavorite.isInvisible = !check
+        binding.buttonUnFavorite.isInvisible = check
+    }
+
+    override fun isInsertedCoinFavorite(long: Long) {
+        context?.showMessage("Insert Success")
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/coinqapp/ui/coindetail/CoinDetailFragment.kt]: hard code

File: app/src/main/java/com/sunasterisk/coinqapp/ui/coindetail/CoinDetailFragment.kt [hunk 2]
```diff
@@ -68,6 +97,30 @@ class CoinDetailFragment : BaseFragment<FragmentCoinDetailBinding>() {
...
+    }
+
+    override fun isDeletedCoinFavorite(boolean: Boolean) {
+        context?.showMessage("Delete Success")
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/coinqapp/ui/coindetail/CoinDetailFragment.kt]: tương tự

File: app/src/main/res/layout/fragment_favorite.xml [hunk 3]
```diff
@@ -1,7 +1,51 @@
...
+        android:id="@+id/toolbar"
+        android:layout_width="0dp"
+        android:layout_height="?attr/actionBarSize"
+        android:background="@color/green_bot_nav"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <TextView
+        android:id="@+id/toolbarTitle"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_favorite.xml]: đặt tên hay thế

File: app/src/main/res/layout/fragment_favorite_coin.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@android:color/white">
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_favorite_coin.xml]: bỏ đi

File: app/src/main/res/layout/fragment_favorite_exchange.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@android:color/white">
+
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_favorite_exchange.xml]: cái nào mà chỉ có 1 view con thì bỏ viewgroup đi cx dk

File: app/src/main/res/layout/item_favorite.xml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            android:contentDescription="@null"
+            app:layout_constraintDimensionRatio="1:1"
+            app:layout_constraintWidth_percent="0.07"
+            android:src="@drawable/ic_splash_screen"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <TextView
+            android:id="@+id/textNameItem"
```
Comment (tuannm-1276) [inline app/src/main/res/layout/item_favorite.xml]:
```suggestion
            android:id="@+id/textName"
```

File: app/src/main/res/layout/fragment_favorite_coin.xml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:layout_marginTop="@dimen/dp_5"
+    android:background="@android:color/white"
+    android:orientation="vertical"
+    app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+    tools:listitem="@layout/item_favorite">
+
+</androidx.recyclerview.widget.RecyclerView>
+
```
Comment (tuannm-1276) [inline app/src/main/res/layout/fragment_favorite_coin.xml]: remove

**Pull request merged**